common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/io/impl/AbstractGeoInstanceWriter.java

```diff
@@ -139,6 +170,63 @@ else if (value instanceof GeometryProperty<?>) {
 	}
 
 	/**
+	 * Returns a pair of unified geometry of given geometry and associated CRS
+	 * definition based on Winding order supplied.
+	 * 
+	 * @param pair A pair of Geometry and CRSDefinition, on which winding
+	 *            process will get done.
+	 * @return Unified Pair .
+	 */
+	protected Pair<Geometry, CRSDefinition> unifyGeometryPair(Pair<Geometry, CRSDefinition> pair) {
+
+		// get Geometry object
+		Geometry geom = pair.getFirst();
+		if (geom == null) {
+			return pair;
+		}
+		// unify geometry
+		geom = unifyGeometry(geom);
+		return new Pair<>(geom, pair.getSecond());
+	}
+
+	/**
+	 * Returns a unified geometry of given geometry based on Winding order
+	 * supplied.
+	 * 
+	 * @param geom The Geometry object, on which winding process will get done.
+	 * @return Unified geometry .
+	 */
+	protected Geometry unifyGeometry(Geometry geom) {
+		if (geom == null) {
+			return geom;
+		}
+		// getting winding order
+		EnumWindingOrderTypes windingOrder = getWindingOrder();
+
+		if (windingOrder == null || windingOrder == EnumWindingOrderTypes.DONTCHANGE) {
+			return geom;
+		}
+		else {
+			Geometry unifiedGeometry;
+			// unify geometry using WindingOrder utility.
+
+			switch (windingOrder) {
+
+			case COUNTERCLOCKWISE:
+				unifiedGeometry = WindingOrder.unifyWindingOrder(geom, true);
+				break;
+			case CLOCKWISE:
+				unifiedGeometry = WindingOrder.unifyWindingOrder(geom, false);
+				break;
+			default:
+				unifiedGeometry = WindingOrder.unifyWindingOrder(geom, true);
```

Comment:
This case should never happen, other than the enum is extended. Thus I would suggest one of two things here:

1. Log an error message and don't touch the geometry *or*
2. Throw an exception

No 1 would be the more robust solution, so I would prefer that.